### PR TITLE
NFS setup for macOS

### PR DESCRIPTION
**Pull request by johnnie502**
For https://github.com/Medology/my-account/issues/76

Extract what's working to the submodules

### CODE REVIEW

File: lib/setup_nfs_functions.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+# Creates a Docker volume
+#
+# Arguments:
+#   1 the volume name
+#   2 the volume path
+#   3 the volume type
+#   4 the volume opts
+#######################################
+createDockerVolume() {
+  local VOLUME_NAME=$1
```
Comment (BKorenek) [inline lib/setup_nfs_functions.sh:36]: var's should be lower case :O

File: lib/setup_nfs_functions.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+# Recreate a Docker volume if the existing volume is mounted from incorrect location.
+#
+# Arguments:
+#   1 the volume name
+#   2 the volume path
+#   3 the volume type
+#   4 the volume opts
+#######################################
+recreateVolumeIfMountFromDifferentFolder() {
+  local VOLUME_NAME=$1
```
Comment (BKorenek) [inline lib/setup_nfs_functions.sh:72]: same here

**Review by BKorenek — APPROVED** [review]
setup_volumes have a buncha caps locked vars which implies global, but in truth they are only used in this file so should be lower case. Approving since it's a super simple change

File: bin/setup_volumes.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    # Config file system exports on macOS Catalina
+    uid=${UID:-"$(id -u "$(whoami)")"}
+    gid=${GID:-"$(id -g "$(whoami)")"}
+    writeToFile 'Export file system' "/System/Volumes/Data -alldirs -mapall=${uid}:${gid} localhost" '/etc/exports'
+
+    echo " - Restarting the NFS server"
+    sudo nfsd restart
+
+    echo "NFS exports ready"
+  fi
```
Comment (Chekote) [inline bin/setup_volumes.sh:31]: I feel like the if/else for initializing the NFS exports should be in a separate function

**Review by Chekote — CHANGES_REQUESTED** [review]

File: bin/setup_volumes.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+else
+  # On linux, create local bind volumes
+  TYPE='none'
+  OPTS='bind'
+fi
```
[the remark this answers is not in the document]
Comment (johnnie502) [reply bin/setup_volumes.sh:18]: This doesn't apply when the function is extracted

**Review by Chekote — APPROVED** [review]

**Pull request merged**